integration-tests/src/test_loop/builder.rs

```diff
@@ -158,7 +158,19 @@ impl TestLoopBuilder {
                 num_concurrent_requests_during_catchup: 1,
             }),
         };
-        client_config.tracked_shards = Vec::new();
+
+        // Configure tracked shards.
+        // * single shard tracking for validators
+        // * all shard tracking for RPCs
+        let bp_num = genesis.config.num_block_producer_seats;
+        let cp_num = genesis.config.num_chunk_producer_seats;
+        let v_num = genesis.config.num_chunk_validator_seats;
+        let validator_num = bp_num.max(cp_num).max(v_num) as usize;
+        if idx < validator_num {
```

Comment:
This is a bit implicit and reliant on some random assumptions. Please let me know if this is not fine and you'd like me to do it more properly and how.

Reply:
I agree, but don't see anything significantly better. Please make names like `bp_num` longer or just use genesis config vars directly.

Reply:
To make it a bit more explicit, can we set the number of validators and non-validators more explicitly? like
```
let TestLoopEnv { mut test_loop, datas: node_datas, tempdir } =
        builder.genesis(genesis).validators(validator_clients).rpc_nodes(rpc_nodes).disable_gc().build();
```
I see nayduck does a similar setup (there you set the numbers though, not the client accounts).

Reply:
But validators are actually set in genesis, so in this way we end up setting them twice...